### PR TITLE
Pass main media video (Youtube) atoms as elements in dotcom-rendering data model

### PR DESCRIPTION
**Pull request by nicl**
## What does this change?

Extract and pass video elements and youtube block elements to Dotcom-rendering (AMP) in data model.

Purely a change to the Dotcom-rendering data model.

(Raising on behalf on others, but I haven't done the work on this, just tested.)

### CODE REVIEW

File: common/app/model/dotcomrendering/pageElements/PageElement.scala [hunk 1]
```diff
@@ -181,9 +189,19 @@ object PageElement {
...
       case Embed => extractEmbed(element).toList
-
-      case Contentatom => element.contentAtomTypeData.map(d => ContentAtomBlockElement(d.atomId)).toList
-
+      case Contentatom => (extractAtom match {
+        case Some(mediaAtom: MediaAtom) =>
+          mediaAtom.activeAssets.headOption.map(asset => {
+            YoutubeBlockElement(
+              mediaAtom.id, //CAPI ID
+              asset.id,
```
Comment (SiAdcock) [inline common/app/model/dotcomrendering/pageElements/PageElement.scala]: For comment consistency:

```suggestion
              asset.id, //YouTube ID
```

File: common/app/model/dotcomrendering/pageElements/PageElement.scala [hunk 2]
```diff
@@ -181,9 +189,19 @@ object PageElement {
...
+              mediaAtom.channelId, //Channel ID
+              mediaAtom.title //Caption
+            )
+          })
+        case Some(atom) => Some(ContentAtomBlockElement(atom.id))
```
Comment (SiAdcock) [inline common/app/model/dotcomrendering/pageElements/PageElement.scala]: Does this mean all content atoms are now supported? Or is there more work to do here?

**Review by SiAdcock — APPROVED** [review]
I'm on board 🚢

File: common/app/model/dotcomrendering/pageElements/PageElement.scala (hunk 2 above)
Comment (nicl) [reply common/app/model/dotcomrendering/pageElements/PageElement.scala]: No unfortunately, looking at it, ContentAtomBlockElement is just defined as:

    case class ContentAtomBlockElement(atomId: String) extends PageElement

which doesn't provide enough for DCR to do anything with it.

Comment (SiAdcock) [reply common/app/model/dotcomrendering/pageElements/PageElement.scala]: Cool, thanks for clarifying. A place to build on, at least.

**Comment by AWare** [comment]
I think this was probably the right way to get video main media working without having to fully model atoms. 

I don't agree with handling atoms individually as elements. For just videos right now it makes sense, but when we add more atoms- and assuming that yet more will be added it will make PageElement really messy. 

I'd prefer it if we had one `PageElement` which would represent any atom, and that could have an `atom: Atom` field. We could then deal with all the special case atoms in a single file which would more immediately show the pattern used in rendering atoms. We could then duplicate this pattern in rendering, and it would substantially reduce the effort of adding new atoms to dotcom rendering. 

In this pattern, adding new atom types will require understanding the global pattern of `PageElement` and the local contexts used in this atom code. Pulling this out into its own file/model will mean that you can implement new atoms with just the context and the `PageElement` pattern will not have to be worked with as a prerequisite. 

Mixing atoms in like this will make things harder for the journalism team- which I believe goes against one of our directives for dcr.

**Pull request merged**